clients/include/blas2/testing_spr2_batched.hpp

```diff
@@ -75,99 +77,115 @@ void testing_spr2_batched_bad_arg(const Arguments& arg)
-                                                       batch_count),
-                                  HIPBLAS_STATUS_INVALID_VALUE);
-            EXPECT_HIPBLAS_STATUS(hipblasSpr2BatchedFn(handle,
-                                                       uplo,
-                                                       N,
-                                                       alpha,
-                                                       dx.ptr_on_device(),
-                                                       incx,
-                                                       dy.ptr_on_device(),
-                                                       incy,
-                                                       nullptr,
-                                                       batch_count),
-                                  HIPBLAS_STATUS_INVALID_VALUE);
+            DAPI_EXPECT(HIPBLAS_STATUS_INVALID_VALUE,
+                        hipblasSpr2BatchedFn,
+                        (handle,
+                         uplo,
+                         N,
+                         alpha,
+                         nullptr,
+                         incx,
+                         dy.ptr_on_device(),
+                         incy,
+                         dA.ptr_on_device(),
+                         batch_count));
+            DAPI_EXPECT(HIPBLAS_STATUS_INVALID_VALUE,
+                        hipblasSpr2BatchedFn,
+                        (handle,
+                         uplo,
+                         N,
+                         alpha,
+                         dx.ptr_on_device(),
+                         incx,
+                         nullptr,
+                         incy,
+                         dA.ptr_on_device(),
+                         batch_count));
+            DAPI_EXPECT(HIPBLAS_STATUS_INVALID_VALUE,
+                        hipblasSpr2BatchedFn,
+                        (handle,
+                         uplo,
+                         N,
+                         alpha,
+                         dx.ptr_on_device(),
+                         incx,
+                         dy.ptr_on_device(),
+                         incy,
+                         nullptr,
+                         batch_count));
+
+            int64_t n_64 = 2147483648; // will rollover to -2147483648 if using 32-bit interface
+            // rocBLAS implementation has alpha == 0 quick return after arg checks, so if we're using 32-bit params,
+            // this should fail with invalid-value
+            // Note that this strategy can't check incx as rocBLAS supports negative. Also depends on implementation so not testing cuBLAS for now
+            DAPI_EXPECT((arg.api & c_API_64) ? HIPBLAS_STATUS_SUCCESS
+                                             : HIPBLAS_STATUS_INVALID_VALUE,
+                        hipblasSpr2BatchedFn,
+                        (handle, uplo, n_64, zero, nullptr, 1, nullptr, 1, nullptr, n_64));
```

Comment:
is batch_count intended to be n_64?

Reply:
Yeah, might as well test batch_count has int64_t simultaneously